src/Facets/PolymerCCTPFacet.sol

```diff
@@ -0,0 +1,161 @@
+
+        if (_tokenMessenger == address(0) || _usdc == address(0) || _polymerFeeReceiver == address(0)) {
+            revert InvalidAddress();
+        }
+
+        TOKEN_MESSENGER = ITokenMessenger(_tokenMessenger);
+        USDC = _usdc;
+        POLYMER_FEE_RECEIVER = payable(_polymerFeeReceiver);
+    }
+
+    /// @notice Bridges USDC via PolymerCCTP
+    /// @param _bridgeData The core bridge data
+    /// @param _polymerData Data specific to PolymerCCTP
+    /// @notice Requires caller to approve the LifiDiamondProxy of the bridge amount + polymerFee
+    function startBridgeTokensViaPolymerCCTP(ILiFi.BridgeData memory _bridgeData, PolymerCCTPData calldata _polymerData)
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validatePolymerData(_bridgeData, _polymerData)
+        onlyAllowSourceToken(_bridgeData, USDC)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+
+        // TODO: Do we need this check if it's always going to be usdc?
+        LibAsset.depositAsset(_bridgeData.sendingAssetId, _bridgeData.minAmount);
+
+        _startBridge(_bridgeData, _polymerData);
+    }
+
+    /// @notice Performs a swap before bridging via PolymerCCTP
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _polymerData Data specific to PolymerCCTP
+    function swapAndStartBridgeTokensViaPolymerCCTP(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        PolymerCCTPData calldata _polymerData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validatePolymerData(_bridgeData, _polymerData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+
+        _startBridge(_bridgeData, _polymerData);
+    }
+
+    /// @dev Contains the business logic for the bridge via PolymerCCTP
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _polymerData Data specific to PolymerCCTP
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        PolymerCCTPData calldata _polymerData
+    ) internal {
+        LibAsset.transferERC20(USDC, POLYMER_FEE_RECEIVER, _polymerData.polymerTokenFee);
+
+        // TODO we don't need to use safe approve here?
+        IERC20(USDC).approve(address(TOKEN_MESSENGER), _bridgeData.minAmount - _polymerData.polymerTokenFee);
+
+        // Need tocheck: can we just use destinationChainID as the normal chain id? and can we just mpass in min Amount as the amountT?
```

Comment:
since we support non EVM chains here and LI.FI uses custom chain IDs for nonEvm chains, we need to add a mapping for all non-EVM chains to their actual chainID

Please have a look at RelayFacet for reference: 
https://github.com/lifinance/contracts/blob/main/src/Facets/RelayFacet.sol

If new non-EVM chains are being added later on, we will need to update the facet and redeploy it.